src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -118,9 +122,17 @@ internal void SetTestLogger(Action<string>? writeLineMessageLogger)
         /// </summary>
         protected internal Solution CreateSolution(SolutionInfo solutionInfo)
         {
-            return new Solution(this, solutionInfo.Attributes);
+            var options = _optionService?.GetForceComputedOptions(solutionInfo.GetProjectLanguages())
+                ?? (SerializableOptionSet)this.CurrentSolution.Options;
+            return CreateSolution(solutionInfo, options);
         }
 
+        /// <summary>
+        /// Create a new empty solution instance associated with this workspace, and with the given options.
+        /// </summary>
+        private Solution CreateSolution(SolutionInfo solutionInfo, SerializableOptionSet options)
```

Comment:
>SolutionInfo [](start = 40, length = 12)

Would it make sense to add `SerializableOptionSet` onto `SolutionInfo`?

Reply:
I thought about it, but I think `SolutionAttributes` contains mostly metadata for the solution, while options and its values represents a part of the `SolutionState`. So, I ended up putting it on the solution state.

Reply:
I see. OK.